core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -867,6 +867,11 @@ private[log] class Cleaner(val id: Int,
                                   stats: CleanerStats): Unit = {
     map.clear()
     val dirty = log.logSegments(start, end).toBuffer
+    val nextSegmentStartOffsets : mutable.Buffer[Long] = if (dirty.nonEmpty) {
```

Comment:
Don't know more idiomatic way to write this in Scala

Reply:
Since it's a mutable buffer, how about assigning it to an empty buffer in the declaration, then appending the offsets from `dirty` as well as the `end` offset in an if block. We could also avoid the `:+` 😁

Reply:
thx @mumrah 
added your suggestion(s)